src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/AsyncIteratorMethodBuilder.cs

```diff
@@ -21,20 +22,25 @@ public struct AsyncIteratorMethodBuilder
         // related to it, so we'd need to replicate all of those optimizations (e.g. storing
         // that box object directly into a Task's continuation field).
 
-        private AsyncTaskMethodBuilder _methodBuilder; // mutable struct; do not make it readonly
+        /// <summary>The lazily-initialized built task.</summary>
+        private Task<VoidTaskResult> m_task; // lazily-initialized: must not be readonly. Debugger depends on the exact name of this field.
```

Comment:
What's the benefit to all of these changes to AsyncIteratorMethodBuilder?  It looks like more code to maintain for little-to-no benefit.

Reply:
> It looks like more code to maintain for little-to-no benefit.

Not sure what you mean? They are all just pass through methods; it just removes the wrapper of `AsyncTaskMethodBuilder` as with this change its not adding anything.

Reply:
> Not sure what you mean?

e.g. all of the duplicated creation code on the PROJECTN code paths

Reply:
I can add another interface and reduce it to a single line https://github.com/dotnet/coreclr/commit/659759456e464b2e666474b8e38bbdcfcad536eb; but that's probably worse.

Let me try something else

Reply:
Is now two lines of code (one being the create of the struct) rather than the one line currently; seem ok?